src/settings_doc/hooks.py

```diff
@@ -0,0 +1,57 @@
+from functools import lru_cache
+from itertools import chain
+from pathlib import Path
+from types import ModuleType
+from typing import Iterable, Tuple, Callable, Set, Any, List
+
+from click import BadParameter
+
+from settings_doc import importing
+
+HOOKS_PREFIX = "settings_doc_"
+
+HOOK_INITIALIZE_ENVIRONMENT = "initialize_environment"
+HOOK_PRE_RENDER = "pre_render"
+
+VALID_HOOKS = {HOOK_INITIALIZE_ENVIRONMENT, HOOK_PRE_RENDER}
+
+
+def get_hook_type_from_name(name: str) -> str:
+    return name[len(HOOKS_PREFIX):]
+
+
+def is_a_hook(name: str, member: Any) -> bool:
+    return name.startswith(HOOKS_PREFIX) and callable(member)
+
+
+@lru_cache()
+def load_hooks_from_module(module: ModuleType) -> Iterable[Tuple[str, Callable]]:
+    collected: Set[Any] = set()
+    hooks: List[Tuple[str, Callable]] = []
```

Comment:
Using a dict instead of a list of tuples will allow applying the hooks directly later without iteration.
```suggestion
    hooks: Dict[str, Callable] = {}
```